examples/seismic/tutorials/16_source_wavefield_reconstruction.ipynb

```diff
@@ -0,0 +1,1069 @@
+{
```

Comment:
Nice tutorial. Some generic comments.

I think it would be beneficial to split and/or extend this tutorial for readability. I would 
- first focus purely on the wavefield itself and show snapshots/movie/... of the forward wavefield and its reconstruction (i.e check the snapshot tutorial for ploting)
- Second detail the gradient computation

And on the into side, It would be beneficial to use slightly more explicit variable names, such as `u` and `u_boundary` or such. I would also completely skip the illumination part. While this is indeed very useful in practice, it adds to the complexity of the tutorial.